src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavaTokenizer.java

```diff
@@ -1273,76 +1255,110 @@ public CommentStyle getStyle() {
-                                return;
-                            }
-                        }
-                    }
+        /**
+         * Trim the first part of the JavaDoc comment.
+         *
+         * @param line line reader
+         *
+         * @return modified line reader
+         */
+        UnicodeReader trimJavadocComment(UnicodeReader line) {
+            line = trimEndOfComment(line);
+            int pos = line.position();
+            line.skipWhitespace();
 
-                    // Skip rest of line
-                    while (isAvailable()) {
-                        switch (get()) {
-                            case '*':
-                                next();
+            if (!line.isAvailable()) {
+                return line;
+            }
 
-                                if (is('/')) {
-                                    return;
-                                }
+            if (line.skip('*') == 0) {
+                line.reset(pos);
+            }
 
-                                break;
-                            case '\r': // (Spec 3.4)
-                            case '\n': // (Spec 3.4)
-                                accept('\r');
-                                accept('\n');
-                                continue forEachLine;
+            return line;
+        }
 
-                            default:
-                                next();
-                                break;
-                        }
-                    } // rest of line
-                } // forEachLine
-                return;
-            } finally {
+        /**
+         * Put the line into the buffer.
+         *
+         * @param line line reader
+         */
+        protected void putLine(UnicodeReader line) {
+            // ignore, overridden in subclass
+        }
+
+        /**
+         * Scan document comment for content.
+         */
+        protected void scanDocComment() {
+            if (!scanned) {
+                deprecatedFlag = false;
                 scanned = true;
+
+                if (!accept("/**")) {
+                    return;
+                }
+
+                skip('*');
+                skipWhitespace();
+
+                if (isEOLN()) {
+                    accept('\r');
+                    accept('\n');
+                }
+
+                while (isAvailable()) {
+                    UnicodeReader line = lineReader();
+                    line = trimJavadocComment(line);
+
+                    // If standalone @deprecated tag
+                    int pos = line.position();
+                    line.skipWhitespace();
+
+                    if (line.accept("@deprecated") &&
+                            (!line.isAvailable() ||
+                                    line.isWhitespace() ||
+                                    line.isEOLN() ||
+                                    line.get() == EOI)) {
+                        deprecatedFlag = true;
+                    }
```

Comment:
As mentioned recently elsewhere, this is generally insufficient, albeit no worse than before.
Generally, we should move away from this code, and eliminate `deprecated Flag` entirely. Deprecation is (or should be) indicated by the `@Deprecated` annotation, with associated documentation given by a plain/regular `@deprecated` tag, as read by `DocCommentParser` or equivalent.